docs/changelog/0.4.0.md

```diff
@@ -0,0 +1,532 @@
+with the most recent versions of Scala 2 available at the moment of release - 2.12.12 and 2.13.4. 
+Scala 2.11 is still support, what more it was enhanced by multiple features, i.e. handling of java default methods.
+
+### JUnit Native Plugin
+Scala Native now comes with JUnit support out of the box, this means that you can write tests in the same way you would do for a Java project.
+To enable JUnit tests all you will need to do is to add the two following lines to your `build.sbt`.
+
+```scala
+addCompilerPlugin("org.scala-native" % "junit-plugin" % SNVersion cross CrossVersion.full)
+```
+
+### NIR changes
+With this major release we're forced to break backward compatibility with previous versions of SN. 
+It means also means that any library published using version 0.4.0-M2 or older would not be compatible to use now.
+
+We also have managed to fix multiple Native Intermiediete Representation related bugs, i.e the way Strings and Characters 
+were stored or adding class literal allowing us to handle proper caching of `java.lang.Class` instances. 
+Moreover, we are now collecting source code positions in NIR instructions, allowing us to add first class debug support in following releases.
+
+### Reflective instantiation
+Since this release you are able to reflectively instantiate definitions marked with the `@EnableReflectiveInstantation` annotation.
+It will also make all of its descendants able to be reflectively instantiated. 
+For definitions marked with this annotation Scala Native plugin would generate definitions of static class initializers 
+and invoke them just before the `main` method in your final program. 
+This step would register `LoadableModuleClass` and `InstantiableClass` instances for objects and classes accordingly allowing 
+them to be accessed later via the provided `scalanative.reflect.Reflect` object methods in your code.
+
+Scala Native does not support full reflection support, although this feature might fix most of the issues that could occour 
+in users code.
+
+```scala
+package x.y.z
+
+@EnableReflectiveInstantation
+sealed trait ReflectiveFoo {
+  val value: String = "foo"
+}
+
+object SingleFoo extends ReflectiveFoo
+
+case class MultipleFoo(times: Int) extends ReflectiveFoo {
+  override val value: String = super.value * times
+}
+
+for {
+  cls <- lookupInstantiatableClass("x.y.z.Foo")
+  ctor <- cls.getConstructor(classOf[Int])
+  obj <- ctor.newInstance(5)
+} yield obj // results in Some(new MultipleFoo(5))
+
+for {
+  cls <- lookupLoadableModule("x.y.z.SingleFoo")
+  obj <- cls.loadModule
+} yield obj // results Some(SingleFoo)
+
+```
+
+### Other
+Multiple name mangling and runtime errors have been fixed. For more details check list of merged PRs
+
+
+## Sbt plugin changes
+### Native Config
+We've decided to deprecate the current spare set of `NativeX` setting keys with single `NativeConfig` object. 
```

Comment:
The formulation "We've decided" is weird in the context of a changelog. Stick to simple forms like "We deprecated [...]." or "[...] is/are now deprecated."